process/interceptors/epochStartMetaBlockInterceptor_test.go

```diff
@@ -0,0 +1,171 @@
+		},
+		{
+			description: "nil num connected peers provider",
+			argsFunc: func() interceptors.ArgsEpochStartMetaBlockInterceptor {
+				args := getArgsEpochStartMetaBlockInterceptor()
+				args.NumConnectedPeersProvider = nil
+				return args
+			},
+			exptectedErr: process.ErrNilNumConnectedPeersProvider,
+		},
+		{
+			description: "nil current network epoch setter",
+			argsFunc: func() interceptors.ArgsEpochStartMetaBlockInterceptor {
+				args := getArgsEpochStartMetaBlockInterceptor()
+				args.CurrentNetworkEpochSetter = nil
+				return args
+			},
+			exptectedErr: process.ErrNilCurrentNetworkEpochSetter,
+		},
+		{
+			description: "consensus percentage lower than 0",
+			argsFunc: func() interceptors.ArgsEpochStartMetaBlockInterceptor {
+				args := getArgsEpochStartMetaBlockInterceptor()
+				args.ConsensusPercentage = -10
+				return args
+			},
+			exptectedErr: process.ErrInvalidEpochStartMetaBlockConsensusPercentage,
+		},
+		{
+			description: "consensus percentage higher than 100",
+			argsFunc: func() interceptors.ArgsEpochStartMetaBlockInterceptor {
+				args := getArgsEpochStartMetaBlockInterceptor()
+				args.ConsensusPercentage = 110
+				return args
+			},
+			exptectedErr: process.ErrInvalidEpochStartMetaBlockConsensusPercentage,
+		},
+		{
+			description: "all constructor parameters ok",
+			argsFunc: func() interceptors.ArgsEpochStartMetaBlockInterceptor {
+				args := getArgsEpochStartMetaBlockInterceptor()
+				return args
+			},
+			exptectedErr: nil,
+			shouldNotErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		esmbi, err := interceptors.NewEpochStartMetaBlockInterceptor(tt.argsFunc())
+		if tt.shouldNotErr {
+			require.NoError(t, err)
+			require.False(t, check.IfNil(esmbi))
+			continue
+		}
+
+		require.True(t, errors.Is(err, tt.exptectedErr))
+		require.True(t, check.IfNil(esmbi))
+	}
+}
+
+func TestEpochStartMetaBlockInterceptor_ProcessReceivedMessageUnmarshalError(t *testing.T) {
+	t.Parallel()
+
+	esmbi, _ := interceptors.NewEpochStartMetaBlockInterceptor(getArgsEpochStartMetaBlockInterceptor())
+	require.NotNil(t, esmbi)
+
+	message := &mock.P2PMessageMock{DataField: []byte("wrong meta block  bytes")}
+	err := esmbi.ProcessReceivedMessage(message, "")
+	require.Error(t, err)
+}
+
+func TestEpochStartMetaBlockInterceptor_EntireFlowShouldWorkAndSetTheEpoch(t *testing.T) {
+	t.Parallel()
+
+	expectedEpoch := uint32(37)
+	wasCalled := false
+	args := getArgsEpochStartMetaBlockInterceptor()
+	args.CurrentNetworkEpochSetter = &mock.CurrentNetworkEpochSetterStub{
+		SetCurrentEpochCalled: func(epoch uint32) {
+			require.Equal(t, expectedEpoch, epoch)
+			wasCalled = true
+		},
+	}
+	args.NumConnectedPeersProvider = &mock.MessengerStub{
+		ConnectedPeersCalled: func() []core.PeerID {
+			return make([]core.PeerID, 6) // 6 connected peers
+		},
+	}
+	args.ConsensusPercentage = 50 // 50% , so at least 3/6 have to send the same meta block
+
+	esmbp, _ := interceptors.NewEpochStartMetaBlockInterceptor(args)
```

Comment:
esmbi instead esmbp just to keep the same pattern in the whole file

Reply:
done